db/database.go

```diff
@@ -142,6 +142,7 @@ type DatabaseContextOptions struct {
 	SGReplicateOptions        SGReplicateOptions
 	SlowQueryWarningThreshold time.Duration
 	QueryPaginationLimit      int // Limit used for pagination of queries. If not set defaults to DefaultQueryPaginationLimit
+	UserXattrKey              *string
```

Comment:
As with config, it looks like it should be fine to store this as string instead of *string (which would also avoid the XattrKeyOrEmpty overhead) - I think the pointer approach is only needed when the empty/default value is valid, and so we need to differentiate between empty and undefined.